src/sagemaker/session.py

```diff
@@ -4249,6 +4253,15 @@ def get_model_package_args(
             "Image": image_uri,
             "ModelDataUrl": model_data,
         }
+        if entry_point is not None:
+            # This is a script mode model and its container def wasn't set
+            # (this happens when an estimator is passed to RegisterModelStep instead
+            # of a model). Set the env vars needed for script mode.
+            env_map = {
+                DIR_PARAM_NAME: "/opt/ml/model/code",
+                SCRIPT_PARAM_NAME: os.path.basename(entry_point),
+            }
+            container["Environment"] = env_map
```

Comment:
Seems we're appending the env vars in `session.py` so I have some concerns on this:
- Once we release the new interface ModelStep and deprecate the RegisterModel, this logic will be redundant
- As we're changing outside of the Pipeline scope, it may break some of the model registry without pipeline use like:
  - For TensorFlowModel, even if `entry_point` is given, it does not append DIR_PARAM_NAME or SCRIPT_PARAM_NAME in the env vars (see [here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/model.py#L337)
  - For Model type,  if only `entry_point` is given without the `source_dir`, the `DIR_PARAM_NAME` is a s3 path. However, here we override it to be a local path. This may not work for the case outside of pipeline scope.
You can find more env cases in [unit tests here](https://github.com/aws/sagemaker-python-sdk/pull/3076/files#diff-2019ff8a82e7a8af31bee2df6c7df55901496e5cb8b8b383f4c7b6a9b051000eR686-R772)

Note: if the link does not work, please check the parameters of `test_request_compare_of_register_model_under_different_sessions` in `test_model_step.py`

Reply:
Have discussed this with Zhankui previously, as the to be deprecated `RegisterModel` is not well defined and it's bothering us with endless patching, can we simply remove its entry for user script input? 
If there're any user reports this, we can direct them to the ModelStep with the public doc prepared by Jerry.

Sorry that I didn't engage you when reviewing the ModelStep given the tight timeline.

Reply:
Those are good callouts; if we go this route we'll need to add logic for all the various arg combinations and frameworks.

I don't think we should break backwards compatibility by removing any args though. The script mode still works in most cases, but it's reasonable to prefer deprecating this and moving to ModelStep instead of spending time fixing (the latest) issue with it.

I see your ModelStep PR is ready for merge. Let's go that route instead of spending more time fixing RegisterModel step.